internal/cluster/openshift.go

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package cluster
+
+import (
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+// Create a new openshift client using the in cluster config
+func openShiftClient() (*configv1client.ConfigV1Client, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if config != nil {
+		client, err := configv1client.NewForConfig(config)
+		if err != nil {
+			return client, err
+		}
+		return client, nil
+	}
+	return nil, nil
```

Comment:
I don't think `rest.InClusterConfig()` returns `nil` if `err` is `nil` -- we shouldn't need the check `config != nil` (or if we do we should return an error here.

This leaves `OpenshiftVersion()` open to a nil error below, since we don't check that `client` is nil.